core/src/main/scala/org/apache/spark/deploy/DependencyUtils.scala

```diff
@@ -61,11 +61,13 @@ private[deploy] object DependencyUtils extends Logging {
       hadoopConf: Configuration,
       secMgr: SecurityManager): String = {
     val targetDir = Utils.createTempDir()
+    val fileSeparator = Pattern.quote(File.separator)
```

Comment:
Mind elaborating why this should be quoted

Reply:
My bad. `separator` is a string and will be interpreted as a regex. Split on `File.separatorChar` instead and it will split as expected. No Pattern needed.

Reply:
The Java String.split method treats its parameter as a regular expression. The File.separator is a single backslash '\' it throws a PatternSyntaxException. Wrapping it in Pattern.quote ensures it is interpreted as a literal value.

Reply:
This should be OK in Scala: 
```
scala> "foo\\bar".split('\\')
res2: Array[String] = Array(foo, bar)
```

Reply:
Changed to use File.separatorChar.